minor changes to build in BSD (Net and Free)

isildur-g opened the pull request:
minor changes to build in BSD (Net and Free)

Commit:
support building on NetBSD

Commit:
updated readme to reflect FreeBSD build

Commit:
lets rather not add lines of code not yet used anywhere

Commit:
minor change for more certain success in netbsd

Commit:
formatting in readme

Commit:
more readme format tinkering

Commit:
lets not disable warnings

Commit:
maybe netbsd is more pedantic about this?

markos commented on README.md:
````diff
@@ -113,6 +113,28 @@ Assuming an existing HomeBrew installation:
+```
+export CC="/usr/pkg/gcc12/bin/cc"
+export CXX="/usr/pkg/gcc12/bin/g++"
+export CXXFLAGS="-gdwarf-4"
````
it would be better to have these flags set automatically within a CMake file, eg cmake/osdetection.cmake.

markos commented on src/util/popcount.h:
```diff
@@ -36,7 +36,12 @@
+#endif
 
+
+#ifndef __NetBSD__
```
this should instead be a HAVE_BUILTIN_POPCOUNT and this flag should be set in CMake/cflags-generic.cmake or similar.

Commit:
incremental improvement in cleanliness

Commit:
another small cleanup in readme

Commit:
moved HAVE_BUILTIN_POPCOUNT def to cmake

Commit:
typo fix

Commit:
shortened freebsd text

Commit:
whitespace editing in readme

markos commented on README.md:
````diff
@@ -113,6 +113,27 @@ Assuming an existing HomeBrew installation:
+export CC="/usr/local/bin/gcc"
+export CXX="/usr/local/bin/g++"
+```
+
````
Could you also add instructions on how to install the required dependencies on FreeBSD and NetBSD?

Commit:
more verbose instructions for preparing BSD systems.

Commit:
also added note for CC/CXX vars in fbsd/ppc which are different.

Commit:
more system prep info for bsd

Commit:
some more bsd detail

Commit:
also sqlite info for bsd

Commit:
slightly clearer comments in netbsd section

Commit:
added copyright info for modified src files

Commit:
adding libpcap to the bsd packages to install for building/running

Commit:
added ccache for freebsd install recommendations

Commit:
tiny edit to readme

Commit:
one more place to fix where clang in bsd is more picky than gcc in linux

Commit:
first try at getting cmake to leave custom shared lib paths with the binary,
with package added compilers we need to find the right std libs from the
compiler we added, not the base install libs.

Commit:
united the static fat runtime dispatcher with the BSD support.

Commit:
probably fixed the bit about not finding the right libs in some bsd installs

Commit:
bit hacky but it works for setting rpath in freebsd

Commit:
made the rpath finding a bit more flexible than just hardcoded gcc12 and gcc13

Commit:
cmake adds newlines to variables set from command output, trashing the
usefulness of the flexible method of setting the rpath var. back to
the clunky manual setting.

Commit:
added a fixme for the clunky rpath setting logic

Commit:
removing the dispatcher.c changes from this branch/PR

markos reviewed (approved):
Great job, Approved!
